modules/data.atmosphere/R/met2CF.csv.R

```diff
@@ -5,245 +5,511 @@
-##'                 CF_name = df$CF_name, 
-##'                 CF_units = df$CF_units,
-##'                 skip=7,
-##'                 header=1,
-##'                 na.strings=c("-9999","-6999","9999"))  
-##'   lat = 40
-##'   lon = -80
-##'   met2CF.csv(in.path,in.file,outfolder,format,lat,lon)
+##' library(PEcAn.DB)
+##' library(lubridate)
+##' library(RPostgreSQL)
+##' bety = list(user="bety", password="bety", host="localhost", dbname="bety", driver="PostgreSQL", write=TRUE)
+##' con <- db.open(bety)
+##' in.path <- '/home/carya/sites/willow/'
+##' in.prefix <- 'FLX_US-WCr_FLUXNET2015_SUBSET_HH_1999-2014_1-1'
+##' outfolder <- '~/'
+##' input.id <- 5000000005
+##' format <- query.format.vars(input.id=input.id,con)
+##' start_date <- ymd_hm('200401010000')
+##' end_date <- ymd_hm('200412312330')
+##' PEcAn.data.atmosphere::met2CF.csv(in.path,in.prefix,outfolder,start_date,end_date,format,overwrite=TRUE)
 ##' }
-met2CF.csv <- function(in.path, in.file, outfolder, format, lat=NULL, lon=NULL, nc_verbose = FALSE,...){
+met2CF.csv <- function(in.path, in.prefix, outfolder, start_date, end_date, format, lat=NULL, lon=NULL, nc_verbose = FALSE, overwrite=FALSE, ...){
+  require(PEcAn.utils)
+  require(lubridate)
+  require(udunits2)  
+  require(ncdf4)
   
+  start_year <- year(start_date)
+  end_year <- year(end_date)
+  if(!file.exists(outfolder)){
+    dir.create(outfolder)
+  }
+  
+  ## set up results output to return
+  rows <- end_year - start_year + 1
+  results <- data.frame(file=character(rows), host=character(rows),
+                        mimetype=character(rows), formatname=character(rows),
+                        startdate=character(rows), enddate=character(rows),
+                        dbfile.name = in.prefix,
+                        stringsAsFactors = FALSE)
+  
+  in.file <- paste0(in.prefix,'.csv')
   files <- dir(in.path, in.file, full.names = TRUE)
   files <- files[grep("*.csv",files)]
   if(length(files)==0){
     return(NULL)
     logger.warn("No met files named ", in.file, "found in ", in.path)
   }
-  dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)  # does nothing if dir exists
+  files <- files[1]
   
-  for(i in 1:length(files)){
-    
-    new.file <- file.path(outfolder, gsub(".csv","_CF.nc",basename(files[i])))
-    
+  # get lat/lon from format.vars if not passed directly
+  if (missing(lat) || is.null(lat)) { lat <- format$lat }
+  if (missing(lon) || is.null(lon)) { lon <- format$lon }
+  
+  #create new filename by swapping .csv with .nc, and adding year strings from start to end year
+  #year(start_date)
+  all_years <- start_year:end_year
+  all_files <- paste0(file.path(outfolder,gsub(".csv","",basename(files))),".",as.character(all_years),".nc")
+  
+  results$file <- all_files
+  results$host <- fqdn()
+  results$startdate <- paste0(all_years,"-01-01 00:00:00")
+  results$enddate <- paste0(all_years,"-12-31 23:59:59")
```

Comment:
What if the file itself only contains a partial year of data? Probably doesn't need to be fixed right now, but something we should keep an eye on.